### PR TITLE
Bluetooth improvements

### PR DESCRIPTION
**Pull request by sebaszm**
Adds:
- async pairing process,
- passkey pairing,
- numeric comparison pairing,
- legacy/pin-code pairing,
- EIR parsing (name, class, supported uuids retrieving),
- local device class setting,
- UUID code extracted to a separate unit,
- adds abort pairing to json-rpc,
- bugfixes.

### CODE REVIEW

**Comment by sebaszm** [comment]
"&copy" flagged by protex :)

File: Source/bluetooth/HCISocket.h [hunk 1]
```diff
@@ -148,6 +149,74 @@ namespace Bluetooth {
...
+        static constexpr uint8_t  EIR_UUID32_SOME = 0x04;
+        static constexpr uint8_t  EIR_UUID32_ALL = 0x05;
+        static constexpr uint8_t  EIR_UUID128_SOME = 0x06;
+        static constexpr uint8_t  EIR_UUID128_ALL = 0x07;
+        static constexpr uint8_t  EIR_NAME_SHORT = 0x08;
+        static constexpr uint8_t  EIR_NAME_COMPLETE = 0x09;
+        static constexpr uint8_t  EIR_CLASS_OF_DEV = 0x0D;
+
+    public:
+        EIR()
```
Comment (pwielders) [inline Source/bluetooth/HCISocket.h:164]: Copy Constructor/Assignment operator? Allowed or not ?

File: Source/bluetooth/HCISocket.h [hunk 2]
```diff
@@ -577,8 +648,8 @@ namespace Bluetooth {
                     ::memcpy(stream, &(_buffer[_offset]), result);
                     _offset += result;
 
-                    // printf ("SEND: ");
-                    // for (uint16_t loop = 0; loop < result; loop++) { printf("%02X:", stream[loop]); } printf("\n");
+                    //printf("SEND: ");cfor (uint16_t loop = 0; loop < result; loop++) { printf("%02X:", stream[loop]); } printf("\n");
+                    printf(_T("HCI command: %X:%03X\n"), cmd_opcode_ogf(OPCODE), cmd_opcode_ocf(OPCODE));
```
Comment (pwielders) [inline Source/bluetooth/HCISocket.h]: I guess we forgot to comment this out :-) Or make it a TRACE_L1 ?

File: Source/bluetooth/HCISocket.h [hunk 3]
```diff
@@ -604,56 +675,60 @@ namespace Bluetooth {
...
 
-                    // printf ("RECEIVE: ");
-                    // for (uint16_t loop = 0; loop < length; loop++) { printf("%02X:", stream[loop]); } printf("\n");
+                    //printf("RECEIVE: "); for (uint16_t loop = 0; loop < length; loop++) { printf("%02X:", stream[loop]); } printf("\n");
 
                     if (hdr->evt == EVT_CMD_STATUS) {
                         const evt_cmd_status* cs = reinterpret_cast<const evt_cmd_status*>(ptr);
-                        if (htobs(cs->opcode) == OPCODE) {
+                        if (btohs(cs->opcode) == OPCODE) {
+                            printf(_T("HCI command status: %X:%03X Status=%d\n"),
```
Comment (pwielders) [inline Source/bluetooth/HCISocket.h]: Comment it out ?

File: Source/bluetooth/HCISocket.h [hunk 4]
```diff
@@ -604,56 +675,60 @@ namespace Bluetooth {
...
                                 _error =  Core::ERROR_GENERAL;
-                                printf(_T(">>EVT_CMD_STATUS: %X-%03X Error: %d\n"), (cs->opcode >> 10) & 0xF, (cs->opcode & 0x3FF), cs->status);
                             }
                             result = length;
                         }
                     } else if (hdr->evt == EVT_CMD_COMPLETE) {
                         const evt_cmd_complete* cc = reinterpret_cast<const evt_cmd_complete*>(ptr);
-                        if (htobs(cc->opcode) == OPCODE) {
+                        if (btohs(cc->opcode) == OPCODE) {
+                            printf(_T("HCI command complete: %X:%03X %s\n"),
```
Comment (pwielders) [inline Source/bluetooth/HCISocket.h]: Comment ?

File: Source/bluetooth/UUID.h [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+    class UUID {
+    private:
+        static const uint8_t BASE[];
+
+    public:
+        UUID() {
+            _uuid[0] = 0;
+        }
+        UUID(const uint16_t uuid)
```
Comment (pwielders) [inline Source/bluetooth/UUID.h:36]: make this and the next one also explicit?

**Review by pwielders — CHANGES_REQUESTED** [review]
Real good addition. Finally a lot of the loose ends are now properly addresses. Just think that it is good to avoid that we start spamming (printf's) usefull for debugging (Hence Why I also typically leave them in the code but mostly commented out. Would like to avoid that we are generating an awfull amount of logging at the customer.

**Review by pwielders — APPROVED** [review]
Since the requested changes are not functional, will be paicked up in the next delivery.

**Pull request merged**